modules/services/easyeffects.nix

```
@@ -35,6 +86,9 @@ in {
         Will likely need to launch easyeffects to initially create preset.
       '';
     };
+
+    extraInputPresets = presetOptionType;
+    extraOutputPresets = presetOptionType;
```

Maybe this could simply be
```suggestion
    extraPresets = presetOptionType;
```
where the "input" or "output" field inside the preset determine whether it is for input or output?

---

That is, the type of this option would be something like `attrsOf presetType` where `presetType` would be the regular JSON type with the additional requirement that it is an object containing either an "input" or "output" field. In practice I think this would look something like

``` nix
presetType = let baseType = attrsOf jsonFormat.type; in baseType // {
  check = v: baseType.check v && elem (attrNames v) == [ ["input"] ["output"] ];
  description = "EasyEffects input or output JSON preset";
}
```

Completely untested so I'm not certain it will work very well.

---

Thanks for the input, i really appreciate it! 
I agree, it makes much more sense to have it in one attribute set, so i've modified it based on your suggestions. 
I'm fairly inexperienced with programming in nix, so this feedback helped a lot. 
Let me know if there is anything that comes to mind